bindings/rust/s2n-tls/src/callbacks.rs

```diff
@@ -126,16 +126,16 @@ pub(crate) trait AsyncCallback {
 /// Use in conjunction with
 /// [config::Builder::set_client_hello_callback](`crate::config::Builder::set_client_hello_callback()`).
 pub trait ClientHelloCallback {
-    fn poll_client_hello(&self, connection: &mut Connection) -> Poll<Result<(), Error>>;
+    fn poll_client_hello(&mut self, connection: &mut Connection) -> Poll<Result<(), Error>>;
```

Comment:
This can't be `&mut self` since `Config` can be shared between multiple threads and mutable access could cause corruption.